### PR TITLE
model: pytorch: Add custom Neural Network & layer support and loss function entrypoints

### PR DESCRIPTION
**Pull request by sakshamarora1**
Fixes: #794 

TODOs:
- [x] Merge master after #838 gets merged
- [x] Add tests for all new changes
- [x] Add documentation and example
- [x] Add last layer addition support in `model/pytorch/dffml_model_pytorch/pytorch_pretrained.py`
- [x] Update flower classification example after merging master
- [x] Make changes in `model.pytorch/dffml_model_pytorch/pytorch_base.py` to support non classification/image processing tasks
- [ ] Add entrypoint support for optimizers and learning rate schedulers in pytorch after their annotations are added in the `PyTorch` source code (probably in their next release as they are working on it right now)

The last 2 points in the list will be their own separate pull requests

I also want to try and find if there's a non-confusing way to connect pre-trained model architectures and custom neural nets after all of the above has been done!

Signed-off-by: sakshamarora1 sakshamarora1001@gmail.com

### CODE REVIEW

File: model/pytorch/dffml_model_pytorch/utils/utils.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    def __init__(self, config):
+        super().__init__(config)
+        self.function = self.LOSS(**self.config._asdict())
+
+    @classmethod
+    def load(cls, class_name: str = None):
+        for name, loss_class in inspect.getmembers(nn, inspect.isclass):
+            if name.endswith("Loss"):
+                if name.lower() == class_name:
+                    return eval(name + "Function")
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/utils/utils.py:72]:
```suggestion
                    return getattr(sys.modules[__name__], name + "Function")
```

File: model/pytorch/dffml_model_pytorch/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from torchvision import transforms
+
+from dffml.base import BaseDataFlowFacilitatorObject
+from dffml.util.entrypoint import base_entry_point, entrypoint
+from .config import make_pytorch_config
+
+
+def create_layer(layer_dict):
+    try:
+        sequential_dict = nn.Sequential()
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/utils/utils.py:14]: Let's add a comment talking about it

File: model/pytorch/dffml_model_pytorch/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        for name, layer in layer_dict.items():
+            parameters = {k: v for k, v in layer.items()}
+            layer_name = parameters.pop("name")
+
+            sequential_dict.add_module(
+                name, getattr(nn, layer_name)(**parameters)
+            )
+        return sequential_dict
+    except AttributeError:
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/utils/utils.py:23]: Let's use `inspect(layer, dict)` or `inspect(layer, str)` instead of `try/except`

File: model/pytorch/dffml_model_pytorch/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    def transform(self, data, size):
+        return transforms.Compose(
+            [
+                transforms.ToPILImage(),
+                transforms.Resize(size),
+                transforms.CenterCrop((size, size)),
+                transforms.ToTensor(),
+                transforms.Normalize(
+                    [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
```
Comment (pdxjohnny) [inline model/pytorch/dffml_model_pytorch/utils/utils.py:43]: Let's add List[float] options for the mean and std parameters to [Normalize](https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.Normalize) within `PyTorchModelConfig` do not append Normalize to the list passed to Compose if either mean or std is None.

**Comment by sakshamarora1** [comment]
@pdxjohnny Should this example be under tutorials instead of usage?
